Remove TOUCH_SLIDER private build-time conditional

psaavedra opened the pull request:
#### 1684c4b32eb8fafaeae863837e0baebb22350400
<pre>
Remove TOUCH_SLIDER private build-time conditional
<a href="https://bugs.webkit.org/show_bug.cgi?id=253639">https://bugs.webkit.org/show_bug.cgi?id=253639</a>

This build-time conditional was originally added on 2012 to give time to
adapt the BlackBerry port to the changes implemented in Bug 88807 [1][2]

[1] <a href="https://bugs.webkit.org/show_bug.cgi?id=88807">https://bugs.webkit.org/show_bug.cgi?id=88807</a>
[2] <a href="https://bugs.webkit.org/show_bug.cgi?id=88807#c34">https://bugs.webkit.org/show_bug.cgi?id=88807#c34</a>

Reviewed by Chris Dumez and Adrian Perez de Castro.

* Source/WebCore/html/RangeInputType.cpp:
(WebCore::RangeInputType::handleTouchEvent):
(WebCore::RangeInputType::hasTouchEventHandler const):
* Source/WebCore/html/RangeInputType.h:
* Source/cmake/WebKitFeatures.cmake:

Canonical link: <a href="https://commits.webkit.org/261592@main">https://commits.webkit.org/261592@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/10086ce3c5f10e2895d5217bf62e6e47540fd71f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/111874 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/21003 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/90/builds/485 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/3634 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/120564 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/115939 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/22352 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/12060 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/86/builds/3384 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/117638 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/16599 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/99751 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/104884 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/98550 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/88/builds/322 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/45573 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/100334 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/13451 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/89/builds/328 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/92571 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/11601 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/13966 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/9766 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/101592 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/19402 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/52330 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/31562 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/8085 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/15934 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/109633 "Built successfully") | 
| | | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/26967 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

clopez commented on Source/WebCore/html/RangeInputType.cpp:
```diff
@@ -172,17 +172,13 @@ void RangeInputType::handleTouchEvent(TouchEvent& event)
 {
     return true;
 }
-#endif
```
I think here instead of removing the conditional it should be replaced with one for: `#if !PLATFORM(IOS_FAMILY)` to match how this function is defined on the header file at `RangeInputType.h`

aperezdc reviewed (approved):
Patch LGTM after applying the suggested changes. Maybe we want anyone from Apple to rubber-stamp this, given it touches cross-platform code.

psaavedra committed:
Remove TOUCH_SLIDER private build-time conditional
https://bugs.webkit.org/show_bug.cgi?id=253639

This build-time conditional was originally added on 2012 to give time to
adapt the BlackBerry port to the changes implemented in Bug 88807 [1][2]

[1] https://bugs.webkit.org/show_bug.cgi?id=88807
[2] https://bugs.webkit.org/show_bug.cgi?id=88807#c34

Reviewed by Chris Dumez and Adrian Perez de Castro.

* Source/WebCore/html/RangeInputType.cpp:
(WebCore::RangeInputType::handleTouchEvent):
(WebCore::RangeInputType::hasTouchEventHandler const):
* Source/WebCore/html/RangeInputType.h:
* Source/cmake/WebKitFeatures.cmake:

Canonical link: https://commits.webkit.org/261592@main
